plugins/modules/inventory_intent.py

```diff
@@ -1916,14 +1885,16 @@ def get_udf_id(self, field_name):
         """
 
         try:
+            udf_id = None
             response = self.dnac._exec(
                 family="devices",
                 function='get_all_user_defined_fields',
                 params={"name": field_name},
             )
             self.log("Received API response from 'get_all_user_defined_fields': {0}".format(str(response)), "DEBUG")
             udf = response.get("response")
-            udf_id = udf[0].get("id")
+            if udf:
+                udf_id = udf[0].get("id")
 
```

Comment:
what is the value of udf_id if udf is not there.. We will return udf_id.. Can you please check it?

Reply:
we are returning udf_id as None then we check udf_id from the api caller place.